velox/vector/DecodedVector.h

```diff
@@ -395,6 +407,8 @@ class DecodedVector {
   // after constant and dictionary vectors have been peeled off.
   const BaseVector* baseVector_ = nullptr;
 
+  VectorPtr baseVectorShared_;
```

Comment:
nit: Can you mention under what conditions this will be populated. Also, add checks in its getter method to ensure those invariants hold